### DISCUSSION TITLE
Simplifying terms differing only by repeated indices in DiracChains

### DISCUSSION THREAD

**Discussion by hongwanliu** [FeynCalc/feyncalc#98] [Q&A] [answered]
Hi, 

I am trying to simplify a large expression containing two Dirac chains. Here is a very basic example of something that I wish could be simplified: 

```
blah = DiracChain[GA[m], DiracIndex[i], DiracIndex[j]] DiracChain[
    GA[m], DiracIndex[a], DiracIndex[b]] - 
  DiracChain[GA[n], DiracIndex[i], DiracIndex[j]] DiracChain[GA[n], 
    DiracIndex[a], DiracIndex[b]]
```
Is there any way I can get FeynCalc to recognize that these two terms should sum to zero? I've tried Calc, Simplify, FullSimplify, DiracSimplify to no avail. Replacing indices will not work for all terms in the large expression I'm actually working with. Thank you!

PS. On a related note, I realize DiracReduce does not act inside a DiracChain.

**Comment by vsht** [comment, marked as answer]
Hi,

you need to use DiracSimplify with the FCCanonicalizeDummyIndices option

```
DiracChain[GA[m], DiracIndex[i], DiracIndex[j]] DiracChain[GA[m], 
   DiracIndex[a], DiracIndex[b]] - 
 DiracChain[GA[n], DiracIndex[i], DiracIndex[j]] DiracChain[GA[n], 
   DiracIndex[a], DiracIndex[b]]
DiracSimplify[%, FCCanonicalizeDummyIndices -> True]
```

cf. also the very last block in

https://feyncalc.github.io/FeynCalcBook/ref/DiracSimplify.html

> PS. On a related note, I realize DiracReduce does not act inside a DiracChain.

This 
```
DiracChain[GA[i1, i2, i3], DiracIndex[a],   DiracIndex[b]] // DiracReduce
```
works without any issues. DiracReduce will not do anything if the expression contains 
D-dimensional Dirac quantities. Which is correct, since in that case no unambiguous
reduction is possible and one should employ evanescent operators on the Lagrangian
level.

**Reply by hongwanliu** [reply]
@vsht Thank you so much for your prompt reply, I really appreciate it. 

That solved my problem, but I've run into another one: how do you force DiracChains to contract repeated DiracIndices? e.g. 

```
DiracChain[SpinorUBar[p1, mchi], DiracIndex[i]] DiracChain[
  SpinorU[p1, mchi], DiracIndex[i]]
```

I would like this expression to contract to the usual 2*mchi. Again, I've tried DiracSimplify, using DIDelta, but to no avail. 

Thanks again, FeynCalc is great!

**Reply by vsht** [reply]
This is because the correct syntax for an unbarred spinor is `DCHN[i, SpinorU[p, m]]`, not the other way around, cf.
https://feyncalc.github.io/FeynCalcBook/ref/DCHN.html 

Hence, the correct input would be

```
DiracChain[SpinorUBar[p1, mchi], DiracIndex[i]] DiracChain[
   DiracIndex[i], SpinorU[p1, mchi]] // DiracSimplify
```
